src/content/app/structural-variants/components/structural-variants-image/StructuralVariantsImage.tsx

```diff
@@ -14,32 +14,97 @@
  * limitations under the License.
  */
 
-import '@ensembl/ensembl-structural-variants/alignments';
-import type { DetailedHTMLProps, HTMLAttributes } from 'react';
+import { useRef, type DetailedHTMLProps, type HTMLAttributes } from 'react';
+import { useNavigate } from 'react-router-dom';
+import '@ensembl/ensembl-structural-variants';
 
 import config from 'config';
 
-import type { VariantAlignments } from '@ensembl/ensembl-structural-variants/alignments';
+import * as urlFor from 'src/shared/helpers/urlHelper';
+
+import type {
+  StructuralVariantsBrowser,
+  Endpoints,
+  ViewportChangePayload
+} from '@ensembl/ensembl-structural-variants';
+import type { GenomicLocation } from 'src/shared/helpers/genomicLocationHelpers';
+
+import styles from './StructuralVariantsImage.module.css';
 
 const CHROMOSOME_LENGTH = 248956422; // length of chromosome 1
-const REGION_NAME = '1';
-const INITIAL_START = 142_500_000;
-const INITIAL_END = 145_500_000;
 
-const chm13T2TGenomeId = '4c07817b-c7c5-463f-8624-982286bc4355';
-const grch38GenomeId = 'a7335667-93e7-11ec-a39d-005056b38ce3';
+const REFERENCE_TRACKS = ['sv-gene', '950a71e1-5229-459c-822f-d104506d24e8'];
+const ALT_TRACKS = ['sv-gene', 'a8691c70-7d68-4322-937d-938affb1b4ea'];
+
+type Props = {
+  referenceGenomeId: string;
+  altGenomeId: string;
+  referenceGenomeLocation: GenomicLocation;
+  altGenomeLocation: GenomicLocation | null;
+};
+
+const StructuralVariantsImage = (props: Props) => {
+  const navigate = useNavigate();
+
+  const onViewportChangeEnd = (event: CustomEvent<ViewportChangePayload>) => {
+    const referenceGenomeLocation = event.detail.reference;
+    const altGenomeLocation = event.detail.alt;
+    const regionName = props.referenceGenomeLocation.regionName;
+
+    const url = urlFor.structuralVariantsViewer({
+      referenceGenomeId: props.referenceGenomeId,
+      altGenomeId: props.altGenomeId,
+      referenceGenomeLocation: {
+        regionName,
+        ...referenceGenomeLocation
+      },
+      altGenomeLocation: {
+        regionName,
+        ...altGenomeLocation
+      }
+    });
+
+    navigate(url, { replace: true });
+  };
+  const onViewportChangeEndRef = useRef(onViewportChangeEnd);
+
+  const onMount = (element: StructuralVariantsBrowser) => {
```

Comment:
Since you're replacing the sv-browser element on genome change, this agent flag might be relevant: React will call the callback ref with null when the `<ens-sv-browser>` unmounts or is replaced. `onMount` assumes a DOM element and immediately calls `element.addEventListener`, so the first unmount will throw `Cannot read properties of null` and the cleanup you return is never invoked (React ignores return values from callback refs). Listeners will leak and the image view will crash whenever the browser element is removed, e.g. when switching to the interstitial view or changing the keyed region. Guard for null and move the add/remove logic into `useEffect`/`useRef`.